src/client/Microsoft.Identity.Client/ManagedIdentity/CsrMetadata.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    /// <summary>
+    /// Represents metadata required for Certificate Signing Request (CSR) operations.
+    /// </summary>
+    internal class CsrMetadata
+    {
+        public string ClientId { get; }  // client_id of the Managed Identity
+        public string TenantId { get; }  // AAD Tenant of the Managed Identity
+        public string Cuid { get; }  // VM unique Id
+        public string AttestationEndpoint { get; }  // MAA Regional / Custom Endpoint for attestation purposes.
+
+        private CsrMetadata(CsrMetadataResponse response)
+        {
+            ClientId = response.ClientId;
+            TenantId = response.TenantId;
+            Cuid = response.Cuid;
+            AttestationEndpoint = response.AttestationEndpoint;
+        }
+
+        /// <summary>
+        /// Tries to create a CsrMetadata instance from a CsrMetadataResponse, logs warnings and returns null if any field is missing.
```

Comment:
I think you'd want to throw an exception instead and bail instead? This is a server bug.

Reply:
Yes, it'd be a server error, but we only want to throw an exception if `probeMode` is set to `true`. We throw an exception in `ImdsV2ManagedIdentitySource` if null was returned and `probeMode` is set to `false`.

Reply:
resolved